File: core/evolution/semantic_change_forecast_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def forecast_semantic_change(
    changes: List[Dict[str, Any]],
) -> Dict[str, Any]:

    ordered = sorted(
        changes,
        key=lambda x: str(
            x.get("id")
        ),
    )

    return {
        "forecast": ordered,
        "forecast_size": len(
            ordered
        ),
    }

File: src/evolution/semanticChangeForecastEngine.ts
/**
 * Converted from Python: core/evolution/semantic_change_forecast_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function forecastSemanticChange(changes: any): any {
  var ordered: any = py.sorted(changes, {key: ((x: any) => py.toStr(py.get(x, "id"))) as (item: any) => any});
  return {"forecast": ordered, "forecast_size": py.len(ordered)};
}
